### PR TITLE
Add logging

### PR DESCRIPTION
**Pull request by JamesEarle**
Move some longer functions to actions instead

### CODE REVIEW

**Review by Will-Hyde — APPROVED** [review]

File: src/actions/getPaymentTokenAllowance.ts [hunk 1]
```diff
@@ -1,16 +1,28 @@
...
+    provider,
+    paymentTokenAddress
+  );
+  const allowance = await paymentToken.allowance(account, zAuctionAddress);
 
-    return allowance;
-  };
\ No newline at end of file
+  logger.trace(
+    `User ${account} has allowance of ${allowance} for payment token ${paymentToken}`
```
Comment (Remscar) [inline src/actions/getPaymentTokenAllowance.ts]: Might be cleaner to turn the big number into an ether value.

File: src/actions/placeBid.ts [hunk 2]
```diff
@@ -29,28 +32,37 @@ export interface PlaceBidActionParameters {
...
-  const paymentToken = await contract.getPaymentTokenForDomain(params.bid.tokenId);
+
+  const contract = await getZAuctionContract(
+    params.config.web3Provider,
+    params.config.zAuctionAddress
+  );
+  const paymentToken = await contract.getPaymentTokenForDomain(
+    params.bid.tokenId
+  );
+  logger.trace(
```
Comment (Remscar) [inline src/actions/placeBid.ts:44]: Why not just combine with the first log statement?

File: src/api/actions/encodeBid.ts [hunk 3]
```diff
@@ -12,14 +15,17 @@ export const encodeBid = async (
   bidParams: BidParameters
 ): Promise<SignableBid> => {
   const uri = `${apiUrl}/bid`;
+
+  logger.trace(`Calling to encode a bid at ${uri} with ${bidParams}`);
```
Comment (Remscar) [inline src/api/actions/encodeBid.ts]: Have you tested this to see what the log message will be? Will it just be `with [Object object]`? or will it actually print the parameters

File: src/actions/placeBid.ts (hunk 2 above)
Comment (JamesEarle) [reply src/actions/placeBid.ts:44]: `paymentToken` doesn't exist until this point but I can restructure so it's one message

**Review by Remscar — APPROVED** [review]

**Pull request merged**